src/components/Editor/Tab.js

```diff
@@ -202,10 +202,7 @@ export default connect(
     return {
       tabSources: getSourcesForTabs(state),
       selectedSource: selectedSource,
-      sourceMetaData: getSourceMetaData(
-        state,
-        selectedSource && selectedSource.get("id")
-      ),
+      getMetaData: sourceId => getSourceMetaData(state, sourceId),
```

Comment:
so the problem with this approach is that it is not memoized

I think we need a new selector

```js
export const getSelectedSourceMetaData = ...
```

Reply:
Do we want the meta data of the selected source?